src/ValidatableExtensions/ValidatableExtensions.StringProperties.cs

```diff
@@ -35,6 +35,36 @@ public static partial class ValidatableExtensions
         return ref validatable;
     }
 
+    /// <summary>
+    /// Throws an exception if the string returned from the given <paramref name="func"/> is null or empty.
+    /// </summary>
+    /// <remarks>
+    /// The default exception thrown is an <see cref="ArgumentException"/>.
+    /// </remarks>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static ref readonly Validatable<TValue> IfNullOrEmpty<TValue>(this in Validatable<TValue> validatable, Func<TValue, string> func, [CallerArgumentExpression("func")] string? funcName = null)
```

Comment:
Perhaps this should be `string?` (H&B)
```suggestion
    public static ref readonly Validatable<TValue> IfNullOrEmpty<TValue>(this in Validatable<TValue> validatable, Func<TValue, string?> func, [CallerArgumentExpression("func")] string? funcName = null)
```